sympy/logic/boolalg.py

```diff
@@ -1869,6 +1869,12 @@ def _rem_redundancy(l1, terms):
 
     return essential
 
+def _binary_from_list_or_number(lon, bits):
+        def tobin(n, bits):
+            return [(n >> k) & 1 for k in reversed(range(0, bits))]
+        if isinstance(lon, int):
+            return tobin(lon, bits)
+        return list(lon)
```

Comment:
it doesn't look like a check is done that the list has `bits` bits in it; that should probably be verified

Reply:
OK! I was thinking that _maybe_ it also secretly supported some sort of generator, since the original code plainly made `list` out of everything. But that may in hindsight be to support tuples etc.

Indeed makes sense!